test/unit-test/FreeRTOS_IP/FreeRTOS_IP_utest.c

```diff
@@ -2249,6 +2249,41 @@ void test_prvAllowIPPacket_IncorrectProtocolChecksum( void )
     TEST_ASSERT_EQUAL( eReleaseBuffer, eResult );
 }
 
+void test_prvAllowIPPacket_LoopbackDest( void )
+{
+    eFrameProcessingResult_t eResult;
+    IPPacket_t * pxIPPacket;
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer;
+    UBaseType_t uxHeaderLength = 0;
+    uint8_t ucEthBuffer[ ipconfigTCP_MSS ];
+    IPHeader_t * pxIPHeader;
+
+    memset( ucEthBuffer, 0, ipconfigTCP_MSS );
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxNetworkBuffer->pucEthernetBuffer = ucEthBuffer;
+    pxIPPacket = ( IPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer;
+    pxIPHeader = &( pxIPPacket->xIPHeader );
+
+    *ipLOCAL_IP_ADDRESS_POINTER = 0xFFFFFFFF;
+
+    pxIPHeader->ucVersionHeaderLength = 0x45;
+
+    pxIPHeader->ulDestinationIPAddress = ipLOOPBACK_ADDRESS;
```

Comment:
Small nit, but the majority of these test cases set this to `pxIPHeader->ulDestinationIPAddress = *ipLOCAL_IP_ADDRESS_POINTER;` maybe consider assigning `ipLOCAL_IP_ADDRESS_POINTER` to  `ipLOOPBACK_ADDRESS` so this line can remain the same?

Reply:
Thanks for the feedback!

I get what you mean. The reason I'm learning towards how I have it is because `ipLOCAL_IP_ADDRESS_POINTER` refers to the host IP address, while `ipLOOPBACK_ADDRESS` refers specifically to the destination loopback address.

Put another way, hosts will never have an assigned address of `ipLOOPBACK_ADDRESS` (reserved by standard), so I think testing with a local address different from the loopback address makes sense. That said, totally fine changing it if you think best; I've run it both ways and the test passes.